### PR TITLE
Updating https cipher list

### PR DESCRIPTION
**Pull request by MSLaguana**
This change enforces that we use TLS1.2, and also explicitly specifies the ciphers that we permit rather than using a node-version-dependent list.

This specific list is a reordering of the one in node v4.1.1.

### CODE REVIEW

File: src/remotebuild/lib/server.ts [hunk 1]
```diff
@@ -238,10 +238,34 @@ class Server {
...
+                  "!MD5",
+                  "!PSK",
+                  "!SRP",
+                  "!CAMELLIA"];
                 var sslSettings = {
                     key: certStore.getKey(),
                     cert: certStore.getCert(),
                     ca: certStore.getCA(),
+                    ciphers: cipherList.join(":"),
+                    secureProtocol: "TLSv1_2_server_method",
```
Comment (digeff) [inline src/remotebuild/lib/server.ts]: According to this: https://nodejs.org/api/tls.html

secureProtocol: The SSL method to use, e.g. SSLv3_method to force SSL version 3. The possible values depend on your installation of OpenSSL and are defined in the constant SSL_METHODS.

What version of OpenSSL are we using? Is it always the same? So we know this value will always work? Or can the user change it somehow?

Comment (MSLaguana) [inline src/remotebuild/lib/server.ts]: Previously we were using whatever the defaults were. With this fix, we force it to be TLS1.2 (aka SSL3.3) which is the most secure option at the moment.

File: src/remotebuild/lib/server.ts [hunk 2]
```diff
@@ -238,10 +238,34 @@ class Server {
                 return Q(certStore);
             }).
             then(function (certStore: HostSpecifics.ICertStore): https.Server {
+              // A slight reordering of the node v4.1.1 cipher list
+              var cipherList = ["ECDHE-ECDSA-AES256-GCM-SHA384",
```
Comment (nisheetjain) [inline src/remotebuild/lib/server.ts]: why is the re-ordering needed? can we use crypto.getCiphers() here instead?

**Comment by nisheetjain** [comment]
:+1:  assuming this works for node 0.10+

File: src/remotebuild/lib/server.ts [hunk 3]
```diff
@@ -51,6 +53,13 @@ class Server {
         var app = express();
         app.use(expressLogger("dev"));
         app.use(errorhandler());
+        if (conf.secure) {
+            app.use(helmet.hsts({
+                maxAge: 1000 * 60 * 60 * 24 * 365, // one year in milliseconds
```
Comment (digeff) [inline src/remotebuild/lib/server.ts]: What do you think of using a small library for these kind of things like https://www.npmjs.com/package/milliseconds ?

File: src/remotebuild/lib/server.ts [hunk 4]
```diff
@@ -51,6 +53,13 @@ class Server {
...
+                includeSubdomains: false,
```
Comment (digeff) [inline src/remotebuild/lib/server.ts]: Can you please add comments explaining why did we choose to set this property to false? (Same for all other properties)

**Comment by MSLaguana** [comment]
To enable ECDHE it looks like we need a different kind of key; for now this is good enough (and better than using the node 0.12.x defaults)

**Pull request merged**